### PR TITLE
Sonic sairedis changes for SAI SRV6 and SAI refpoint update to v1.9.0

### PR DESCRIPTION
**Pull request by kperumalbfn**
- Update SAI SEGMENTROUTE object to SAI SRV6 object
- Serialize and deserialize APIs for SRV6 MY_SID_ENTRY object
- SAI refpoint update to V1.9.0

### CODE REVIEW

**Comment by kperumalbfn** [comment]
@kcudnik please check this PR for SAI refpoint update and other SRV6 changes.

File: meta/Meta.cpp [hunk 1]
```diff
@@ -439,6 +439,46 @@ sai_status_t Meta::remove(
...
+    status = m_implementation->remove(my_sid_entry);
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_DEBUG("remove status: %s", sai_serialize_status(status).c_str());
+    }
+    else
+    {
+        SWSS_LOG_ERROR("remove status: %s", sai_serialize_status(status).c_str());
+    }
```
Comment (kcudnik) [inline meta/Meta.cpp:472]: MATA_LOG_STATUS("remove") same for create set and get

File: meta/SaiSerialize.cpp [hunk 2]
```diff
@@ -2409,6 +2482,7 @@ static void sai_deserialize_qos_map_params(
     params.prio           = j["prio"];
     params.pg             = j["pg"];
     params.queue_index    = j["qidx"];
+    params.fc             = j["fc"];
```
Comment (kcudnik) [inline meta/SaiSerialize.cpp:2485]: to be backward compatible you need to check if fc field exists, similar to field mpls_exp below

Comment (kcudnik) [reply meta/SaiSerialize.cpp:2485]: then you probably dont need to modify dumpX.json files

File: syncd/NotificationQueue.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 extern "C" {
 #include <sai.h>
+#include<saimetadata.h>
```
Comment (kcudnik) [inline syncd/NotificationQueue.h:5]: why sudennly this is needed ? also space after include

File: syncd/Syncd.cpp [hunk 4]
```diff
@@ -1004,6 +1004,28 @@ sai_status_t Syncd::processBulkCreateEntry(
...
+            }
+
+            status = m_vendorSai->bulkCreate(
+                    object_count,
+                    entries.data(),
+                    attr_counts.data(),
+                    attr_lists.data(),
+                    mode,
+                    statuses.data());
+
```
Comment (kcudnik) [inline syncd/Syncd.cpp:1025]: unnecessary empty lines, actually in multiple places i noticed that

File: syncd/Syncd.cpp [hunk 5]
```diff
@@ -1004,6 +1004,28 @@ sai_status_t Syncd::processBulkCreateEntry(
         }
         break;
 
+        case SAI_OBJECT_TYPE_MY_SID_ENTRY:
+        {
+            std::vector<sai_my_sid_entry_t> entries(object_count);
+            for (uint32_t it = 0; it < object_count; it++)
```
Comment (kcudnik) [inline syncd/Syncd.cpp:1010]: plesae add empty line before each for in entire PR

File: unittest/meta/TestSaiSerialize.cpp [hunk 6]
```diff
@@ -1063,6 +1063,7 @@ TEST(SaiSerialize, serialize_qos_map)
     EXPECT_EQ(l.key.queue_index, 6);
     EXPECT_EQ(l.key.color, SAI_PACKET_COLOR_RED);
     EXPECT_EQ(l.key.mpls_exp, 0);
+    EXPECT_EQ(l.key.fc, 0);
```
Comment (kcudnik) [inline unittest/meta/TestSaiSerialize.cpp:1066]: you could actually specify some non zero value for better check

File: unittest/meta/TestSaiSerialize.cpp [hunk 7]
```diff
@@ -1072,6 +1073,7 @@ TEST(SaiSerialize, serialize_qos_map)
     EXPECT_EQ(l.value.queue_index, 66);
     EXPECT_EQ(l.value.color, SAI_PACKET_COLOR_GREEN);
     EXPECT_EQ(l.value.mpls_exp, 0);
+    EXPECT_EQ(l.value.fc, 0);
```
Comment (kcudnik) [inline unittest/meta/TestSaiSerialize.cpp:1076]: you could actually specify some non zero value for better check

File: unittest/vslib/test_sai_vs_srv6.cpp [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+TEST(libsaivs, srv6)
+{
+    sai_srv6_api_t *api = nullptr;
+
+    sai_api_query(SAI_API_SRV6, (void**)&api);
+
+    EXPECT_NE(api, nullptr);
+
+    sai_my_sid_entry_t id ;
```
Comment (kcudnik) [inline unittest/vslib/test_sai_vs_srv6.cpp:17]: remove space before ;

File: vslib/sai_vs_nexthopgroup.cpp [hunk 9]
```diff
@@ -13,4 +14,5 @@ const sai_next_hop_group_api_t vs_next_hop_group_api = {
 
     vs_bulk_create_next_hop_group_members,
     vs_bulk_remove_next_hop_group_members,
+    VS_GENERIC_QUAD_API(next_hop_group_map)
```
Comment (kcudnik) [inline vslib/sai_vs_nexthopgroup.cpp:17]: add also unittests for those nae apis in unittest/vslin/test_sai_vs_nexthopgroup.cpp

**Review by kcudnik — CHANGES_REQUESTED** [review]

File: syncd/NotificationQueue.h (hunk 3 above)
Comment (kperumalbfn) [reply syncd/NotificationQueue.h:5]: All SAI_SWITCH_NOTIFICATION macros are defined in saimetadata.h. Earlier these were defined in lib/sairediscommon.h

File: meta/SaiSerialize.cpp (hunk 2 above)
Comment (kperumalbfn) [reply meta/SaiSerialize.cpp:2485]: Thanks, need to revert my change to dumpX.json files as that is causing more problems.

File: syncd/Syncd.cpp (hunk 5 above)
Comment (kperumalbfn) [reply syncd/Syncd.cpp:1010]: Format for MY_SID_ENTRY case is similar to other objects.

File: syncd/Syncd.cpp (hunk 4 above)
Comment (kperumalbfn) [reply syncd/Syncd.cpp:1025]: Format for MY_SID_ENTRY case is similar to other objects.

File: syncd/Syncd.cpp (hunk 5 above)
Comment (kcudnik) [reply syncd/Syncd.cpp:1010]: please fix

File: syncd/Syncd.cpp (hunk 4 above)
Comment (kcudnik) [reply syncd/Syncd.cpp:1025]: please fix

File: meta/SaiSerialize.cpp (hunk 2 above)
Comment (kperumalbfn) [reply meta/SaiSerialize.cpp]: Done

File: syncd/Syncd.cpp (hunk 4 above)
Comment (kperumalbfn) [reply syncd/Syncd.cpp]: Done

File: unittest/meta/TestSaiSerialize.cpp (hunk 6 above)
Comment (kperumalbfn) [reply unittest/meta/TestSaiSerialize.cpp]: Done

File: unittest/vslib/test_sai_vs_srv6.cpp (hunk 8 above)
Comment (kperumalbfn) [reply unittest/vslib/test_sai_vs_srv6.cpp]: Done

File: vslib/sai_vs_nexthopgroup.cpp (hunk 9 above)
Comment (kperumalbfn) [reply vslib/sai_vs_nexthopgroup.cpp:17]: Done

File: syncd/Syncd.cpp (hunk 5 above)
Comment (kperumalbfn) [reply syncd/Syncd.cpp:1011]: Done

**Comment by kperumalbfn** [comment]
@kcudnik most of the comments are addressed. Please check.

**Comment by kperumalbfn** [comment]
@rlhui @kcudnik All sanities are passing for Sonic SAI v1.9.0 upgrade. Could you approve and merge the changes?

**Review by kcudnik — APPROVED** [review]

**Comment by kcudnik** [comment]
@kperumalbfn you recently updated MetaKeyHasher.cpp file adding sid_entry support, pleas add unittests for that file under /sonic-sairedis/unittest/meta/TestMetaKeyHasher.cpp, there are examples in this file for other entries as well so you can copy cat that

**Comment by kperumalbfn** [comment]
@kcudnik will add the UT in the next commit

**Comment by kcudnik** [comment]
> @kcudnik will add the UT in the next commit

Actually i already added them, but you can double check if i missed something

**Comment by kperumalbfn** [comment]
Thanks @kcudnik

**Pull request merged**